### PR TITLE
Deactivate unused bucket:

### PR DESCRIPTION
**Pull request by grandizzy**
- on remove quote token
- on claim collateral
Unused bucket is an intiliazed bucket with no deposit, no debt, no outstanding LP, no collateral.
Deactivate bucket implies updating pointers of upper and lower buckets and deleting from buckets mapping.
Update HPB on claimCollateral

### CODE REVIEW

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -369,6 +385,12 @@ contract ERC20Pool is IPool, Clone, Interest {
         borrower.debt                = 0;
         borrower.collateralDeposited -= requiredCollateral;
 
+        // update HPB
```
Comment (MikeHathaway) [inline src/ERC20Pool.sol:388]: Do we need to update HPB in `repay()` as well?

Comment (grandizzy) [reply src/ERC20Pool.sol:388]: I don't think so as repay happens at lup but I can be wrong, @EdNoepel wdyt?

File: src/libraries/Buckets.sol [hunk 2]
```diff
@@ -91,30 +91,29 @@ library Buckets {
...
+    /// @param bucket_ The price bucket from which collateral should be claimed
     /// @param amount_ The amount of collateral tokens to be claimed
     /// @param lpBalance_ The claimers current LP balance
     /// @return lpRedemption_ The amount of LP tokens that will be redeemed
     function claimCollateral(
         mapping(uint256 => Bucket) storage buckets_,
-        uint256 price_,    // WAD
-        uint256 amount_,   // WAD
+        Bucket storage bucket_,
+        uint256 amount_, // RAY
```
Comment (MikeHathaway) [inline src/libraries/Buckets.sol]: Isn't collateral `amount_` precision set to WAD now?

Comment (grandizzy) [reply src/libraries/Buckets.sol]: Yeah, I think was mistakenly changed when I merged an older patch, will change back to wad

**Review by MikeHathaway — APPROVED** [review]
Logic LGTM. Had some minor comments.

File: src/libraries/Buckets.sol (hunk 2 above)
Comment (grandizzy) [reply src/libraries/Buckets.sol]: updated with https://github.com/ajna-finance/contracts/pull/108/commits/f97042c7faa6ccd157423b919869cc21bcbcffdf

File: src/ERC20Pool.sol (hunk 1 above)
Comment (EdNoepel) [reply src/ERC20Pool.sol:388]: No.  Even if HPB == LUP,  lender would still need to withdraw before HPB could change.

**Review by EdNoepel — APPROVED** [review]
LGTM.  I ran a real-world test on this branch, which didn't reveal any gas consumption concerns.

**Pull request merged**